WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewModel.swift

```diff
@@ -75,45 +80,126 @@ final class ShippingLabelAddressFormViewModel: NSObject {
         }
     }
 
-    var sections: [Section] {
+    func updateSections() {
         var rows: [Row] = [.name, .company, .phone, .address, .address2, .city, .postcode, .state, .country]
-        if addressValidationError?.addressError != nil {
-            if let addressIndex = rows.firstIndex(where: { $0 == .address }) {
-                rows.insert(.fieldError, at: rows.index(after: addressIndex))
+
+        let localErrors = validateAddressLocally()
+
+        if localErrors.contains(.name) {
+            if let index = rows.firstIndex(where: { $0 == .name }) {
+                rows.insert(.fieldError(.name), at: rows.index(after: index))
+            }
+        }
+        else if addressValidationError?.addressError != nil || localErrors.contains(.address) {
+            if let index = rows.firstIndex(where: { $0 == .address }) {
+                rows.insert(.fieldError(.address), at: rows.index(after: index))
             }
         }
-        else {
-            rows.removeAll(where: { $0 == .fieldError })
+        else if localErrors.contains(.city) {
+            if let index = rows.firstIndex(where: { $0 == .city }) {
+                rows.insert(.fieldError(.city), at: rows.index(after: index))
+            }
+        }
+        else if localErrors.contains(.postcode) {
+            if let index = rows.firstIndex(where: { $0 == .postcode }) {
+                rows.insert(.fieldError(.postcode), at: rows.index(after: index))
+            }
+        }
+        else if localErrors.contains(.state) {
+            if let index = rows.firstIndex(where: { $0 == .state }) {
+                rows.insert(.fieldError(.state), at: rows.index(after: index))
+            }
         }
-        return [Section(rows: rows)]
+        else if localErrors.contains(.country) {
+            if let index = rows.firstIndex(where: { $0 == .country }) {
+                rows.insert(.fieldError(.country), at: rows.index(after: index))
+            }
+        }
+        sections = [Section(rows: rows)]
```

Comment:
Is it possible to not have these on `if else` conditions?
I tested the feature and its a bit annoying to have the name as an error, fix it, tap done, and then have another field with an error.

It will be nice to get all errors at once, what do you think?

Reply:
You are probably right, I wasn't sure of the behavior because on the design we always have the case with just one field on error state. I'll change it 👍

Reply:
Definitely better 👍 
<img src="https://user-images.githubusercontent.com/495617/111496412-96a53c00-8740-11eb-8018-5cac6407beab.png" width=300 />